content/specs/auth.md

```diff
@@ -0,0 +1,389 @@
+
+#### Proof Key for Code Exchange (PKCE)
+
+PKCE is mandatory for all Authorization Requests. Clients must generate new, unique, random challenges for every authorization request.
+
+The `S256` challenge method must be supported by all clients and Authorization Servers; see [RFC 7636](https://datatracker.ietf.org/doc/html/rfc7636) for details. The `plain` method is not allowed. Additional methods may in theory be supported if both client and server support them.
+
+#### Pushed Authorization Requests (PAR)
+
+Authorization Servers must support PAR and clients of all types must use PAR for Authorization Requests.
+
+Authorization Servers must set `require_pushed_authorization_requests` to `true` in their server metadata document and include a valid URL in `pushed_authorization_request_endpoint`. See [RFC 9207](https://datatracker.ietf.org/doc/html/rfc9207) for requirements on this URL.
+
+Clients make an HTTPS POST request to the `pushed_authorization_request_endpoint` URL, with the request parameters in the form-encoded request body. They receive a `request_uri` (not to be confused with `redirect_uri`) in the JSON response object. When they redirect the user to the authorization endpoint (`authorization_endpoint`), they omit most of the request parameters they already sent and include this `redirect_uri` along with `client_id` as query parameters instead.
+
+:::note
+PAR is a relatively new and less-supported standard, and the requirement to use PAR may be relaxed if it is found to be too onerous a requirement for client implementations. In that case, Authorization Servers would be required to support both PAR and non-PAR requests with PAR being optional for clients.
+:::
+
+#### Confidential Client Authentication
+
+Confidential clients authenticate themselves during the Authorization Request using a JWT client assertion. Authorization Servers may grant confidential clients longer token/session lifetimes. See "Tokens" section for more context.
+
+The client assertion type to use is `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`, as described in "JSON Web Token (JWT) Profile for OAuth 2.0 Client Authentication and Authorization Grants" ([RFC 7523](https://datatracker.ietf.org/doc/html/rfc7523)). Clients and Authorization Servers must support the `ES256` cryptographic system. The set of recommended systems/algorithms is expected to evolve over time.
+
+Additional requirements:
+
+- confidential clients publish one or more client authentication keys (public key) in the client metadata. This can be either direct JWK format as JSON in the `jwks` field, or as a separate JWKS JSON object on the web linked by a `jwks_uri` URL. A `jwks_uri` URL must be a valid fully qualified URL with `https://` scheme.
```

Comment:
NIT: _must_ publish

Reply:
felt a bit weird to phrase this as a "must", as this is basically what makes a client a confidential client? "confidential clients are those which publish ..."

Reply:
What makes a client confidential is that fact that it uses `urn:ietf:params:oauth:client-assertion-type:jwt-bearer` during auth requests.

A client could declare a jwks/jwks_uri and decide not to use it for some auth request.